### PR TITLE
Include triggerer health status in Airflow /health endpoint

### LINKED ISSUE #31522: `/health` endpoint missed when adding triggerer health status reporting
### Apache Airflow version

main (development)

### What happened

https://github.com/apache/airflow/pull/27755 added the triggerer to the rest api health endpoint, but not the main one served on `/health`.

### What you think should happen instead

As documented [here](https://airflow.apache.org/docs/apache-airflow/2.6.1/administration-and-deployment/logging-monitoring/check-health.html#webserver-health-check-endpoint), the `/health` endpoint should include triggerer info like shown on `/api/v1/health`.

### How to reproduce

Compare `/api/v1/health` and `/health` responses.

### Operating System

mac os

### Versions of Apache Airflow Providers

_No response_

### Deployment

Virtualenv installation

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by pankajkoti**
PR https://github.com/apache/airflow/pull/27755 introduced sending triggerer health status in the `/api/v1/health` 
endpoint and also updated relevant docs but we have the primary `/health` 
endpoint too which is missing this information. The PR addresses this missing 
status report for triggerer health in the `/health` endpoint. It also attempts to 
deduplicate the code between those endpoints so that in future when we need 
to make necessary changes in only one place and at the same time ensure that 
change made in one endpoint is not missed for the other endpoint serving the 
same purpose and thus ensuring consistency in the responses.

fixes: #31522

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by pankajkoti** [comment]
API responses now:
Screenshot 2023-05-25 at 3 28 24 AM

Screenshot 2023-05-25 at 3 28 02 AM
Screenshot 2023-05-25 at 3 27 47 AM

**Review by uranusjr — APPROVED** [review]

**Review by pierrejeambrun — COMMENTED** [review]
Sweet! I could use that also in my Cluster Activity page to display the triggerer health :tada:

Maybe we can add a small test case ? (most of the utils are tested in `test_utils`)

File: airflow/utils/airflow_health.py
Comment (jedcunningham) [inline airflow/utils/airflow_health.py:1]: Can we move this into `airflow/api/common/airflow_health.py` instead?

Comment (pankajkoti) [reply airflow/utils/airflow_health.py:1]: yes, I did feel that that directory prefix (airflow/api/common) was better, but looking at the existing files there was not sure if it is the right place. I will try moving it there now and also add few tests as suggested by @pierrejeambrun 👍🏽

Comment (jedcunningham) [reply airflow/utils/airflow_health.py:1]: Yeah, I do think that's a better home than utils. Sounds good, thanks!

Comment (pankajkoti) [reply airflow/utils/airflow_health.py:1]: moved it and verified both the endpoints work well.

File: airflow/api/common/airflow_health.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from __future__ import annotations
+
+from airflow.jobs.scheduler_job_runner import SchedulerJobRunner
+from airflow.jobs.triggerer_job_runner import TriggererJobRunner
+
+HEALTHY = "healthy"
+UNHEALTHY = "unhealthy"
+
+
+def get_airflow_health():
```
Comment (pierrejeambrun) [inline airflow/api/common/airflow_health.py]: Maybe we could add a return type annotation. (`TypedDict`, or `Dict` at least ?)

**Review by pierrejeambrun — APPROVED** [review]
Minor suggestion but LGTM.

File: airflow/api/common/airflow_health.py (hunk 1 above)
Comment (pankajkoti) [reply airflow/api/common/airflow_health.py]: I tried to be descriptive by specifying the return type as `dict[str, dict[str, str | None]]`, but it failed both locally (I thought my breeze might be acting weird) and also on CI with the below error, so changed it to `dict[str, Any]` which breeze accepted. So, hopefully CI will accept it too 🤞🏽 .

Screenshot 2023-05-27 at 12 04 31 AM

**Review by jedcunningham — APPROVED** [review]

**Comment by pankajkoti** [comment]
1 failing check in CI :( I think this is unrelated, but I will try another rebase to re-trigger CI.

Screenshot 2023-05-27 at 12 59 45 AM

File: airflow/api/common/airflow_health.py (hunk 1 above)
Comment (pankajkoti) [reply airflow/api/common/airflow_health.py]: yes, CI is green now.

Comment (jedcunningham) [reply airflow/api/common/airflow_health.py]: I think you missed a nested level of dict, but I think any is acceptable here too.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**